src/components/grid/grid.test.js

```diff
@@ -0,0 +1,360 @@
+import * as React from "react";
+import GridContainer from "./grid-container/grid-container.component";
+import GridItem from "./grid-item/grid-item.component";
+import { pod, gridPod, gridComponent } from "../../../cypress/locators/grid";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+
+const ViewportSize = (sizeOfViewport) => {
+  switch (sizeOfViewport) {
+    case "default":
+      cy.viewport(1958, 900);
+      break;
+    case "extra small":
+      cy.viewport(599, 900);
+      break;
+    case "small":
+      cy.viewport(959, 900);
+      break;
+    case "medium":
+      cy.viewport(1259, 900);
+      break;
+    case "large":
+      cy.viewport(1920, 900);
+      break;
+    case "extra large":
+      cy.viewport(1922, 900);
+      break;
+    default:
+      throw new Error(
+        `${sizeOfViewport} is not defined in a scope. We want to resize to only for types of viewport`
+      );
+  }
+};
+
+const GridComponent = ({ ...props }) => {
+  return (
+    <>
+      <GridContainer {...props}>
+        <GridItem alignSelf="stretch" justifySelf="stretch" {...props}>
+          <pod alignTitle="left" border padding="medium" variant="primary">
+            1
+          </pod>
+        </GridItem>
+        <GridItem alignSelf="stretch" justifySelf="stretch">
+          <pod alignTitle="left" border padding="medium" variant="primary">
+            2
+          </pod>
+        </GridItem>
+        <GridItem alignSelf="stretch" justifySelf="stretch">
+          <pod alignTitle="left" border padding="medium" variant="primary">
+            3
+          </pod>
+        </GridItem>
+      </GridContainer>
+    </>
+  );
+};
+
+context("Testing Grid component", () => {
+  describe("should render Grid component", () => {
+    it.each([
+      [1, "auto", "auto"],
+      [2, "1", "13"],
+      [3, "1", "13"],
+    ])(
+      "should render Grid viewport to default and verify pod width and height",
+      (index, colStart, colEnd) => {
+        CypressMountWithProviders(<GridComponent />);
+
+        ViewportSize("default");
+        if (index === 1) {
+          gridPod(index).should("have.css", "grid-row", "auto / auto");
+          gridPod(index).should(
+            "have.css",
+            "grid-column",
+            `${colStart} / ${colEnd}`
+          );
+        } else {
+          gridPod(index).should("have.css", "grid-row", "auto / auto");
+          gridPod(index).should(
+            "have.css",
+            "grid-column",
+            `${colStart} / ${colEnd}`
+          );
+        }
+      }
+    );
+
+    it.each([
+      [1, "auto", "auto"],
+      [2, "1", "13"],
+      [3, "1", "13"],
+    ])(
+      "should render Grid viewport to small and verify pod width and height",
```

Comment:
Should this read "extra small"? Also, should there then be a case for checking "small" after the "extra small" one?